src/analyzer/analyzerwaveform.cpp

```diff
@@ -75,6 +82,11 @@ bool AnalyzerWaveform::initialize(TrackPointer tio, int sampleRate, int totalSam
         m_waveformSummary = WaveformPointer(new Waveform(
                 sampleRate, totalSamples, mainWaveformSampleRate,
                 summaryWaveformSamples));
+        //I force it to NOT dirty, because other Mixxx threads like the Main thread can trigger 
+        //TrackDAO.updateTrack(), which would try to save an empty or incomplete analysis. 
+        m_waveform->setDirty(false);
```

Comment:
Unfortunately this can still happen, if the Waveform was dirty before and the Main thread checks it just before it was set to false here.
Are there actually cases where this can be an issue? Maybe we either fix this case, or protect the writing and analyzing by a Mutex using tryLock().